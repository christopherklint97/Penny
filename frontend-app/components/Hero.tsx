interface HeroProps {
  title: string
  text: string
  children?: any
}

export default function Hero({ title, text, children }: HeroProps) {
  return (
    <div>
      <h1 className="text-2xl text-center py-3 mt-5 font-bold">{title}</h1>
      <p className="text-center pb-3">{text}</p>
      <div className="text-center">{children}</div>
    </div>
  )
}
