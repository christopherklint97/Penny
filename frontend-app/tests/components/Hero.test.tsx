import React from 'react'
import { render } from '../testUtils'
import Hero from '../../components/Hero'

describe('Hero component', () => {
  it('renders without crashing', function () {
    render(<Hero title="This is a test" text="Body" />)
  })

  it('matches snapshot', () => {
    const { asFragment } = render(
      <Hero title="This is a test" text="Body" />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
