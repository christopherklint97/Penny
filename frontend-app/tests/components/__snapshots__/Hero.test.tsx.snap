// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="text-2xl text-center py-3 mt-5 font-bold"
    >
      This is a test
    </h1>
    <p
      class="text-center pb-3"
    >
      Body
    </p>
    <div
      class="text-center"
    />
  </div>
</DocumentFragment>
`;
